File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    // 1. sort 로 배열 정렬
    participant.sort();
    completion.sort();
    
    for(let i=0; i<participant.length; i++) {
        // 2. 참가자 배열과 완주자 배열이 같지 않으면 바로 return
        if(participant[i] !== completion[i]) {
            return participant[i];
        }
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        Map<String, Integer> maraMap = new HashMap<>();
        
        // hashmap 에 이름을 key, 동명이인 고려해서 카운트를 value 값으로
        for(String comp : completion) {
            if(maraMap.containsKey(comp)) {
                maraMap.replace(comp , (maraMap.get(comp)) + 1);
            } else {
                maraMap.put(comp, 1);
            }
        }
        
        // participant 에 해당하는 이름에 대한 카운트 -1 해주기(0값이 아닌 경우에만)
        // 만약 해당 조건이 충족되지 않는 경우 return
        for(String part : participant) {
            if(maraMap.containsKey(part) && maraMap.get(part) != 0) {
                maraMap.replace(part , (maraMap.get(part)) - 1);
            } else {
                answer = part;
            }
        }
        return answer;
    }
}